src/pages/iou/MoneyRequestEditWaypointPage.js

```diff
@@ -0,0 +1,51 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import {withOnyx} from 'react-native-onyx';
+import WaypointEditor from './WaypointEditor';
+import ONYXKEYS from '../../ONYXKEYS';
+
+const propTypes = {
+    /** The transactionID of this request */
+    transactionID: PropTypes.string,
+
+    /** Route params */
+    route: PropTypes.shape({
+        params: PropTypes.shape({
+            /** IOU type */
+            iouType: PropTypes.string,
+
+            /** Index of the waypoint being edited */
+            waypointIndex: PropTypes.string,
+        }),
+    }),
+};
+
+const defaultProps = {
+    transactionID: '',
+    route: {
+        params: {
+            iouType: '',
+            waypointIndex: '',
+        },
+    },
+};
+
+// This component is responsible for grabbing the transactionID from the IOU key
+// You can't use Onyx props in the withOnyx mapping, so we need to set up and access the transactionID here, and then pass it down so that WaypointEditor can subscribe to the transaction.
+function MoneyRequestEditWaypointPage({transactionID, route}) {
+    return (
+        <WaypointEditor
+            transactionID={transactionID}
+            route={route}
+        />
+    );
+}
+
+MoneyRequestEditWaypointPage.displayName = 'MoneyRequestEditWaypointPage';
+MoneyRequestEditWaypointPage.propTypes = propTypes;
+MoneyRequestEditWaypointPage.defaultProps = defaultProps;
+export default withOnyx({
+    // We must provide a default value for transactionID here, otherwise the component won't mount
+    // because withOnyx returns null until all the keys are defined
+    transactionID: {key: ONYXKEYS.DISTANCE_REQUEST, selector: (distanceRequest) => (distanceRequest && distanceRequest.transactionID) || ''},
```

Comment:
Instead of doing this stuff with the DISTANCE_REQUEST key, can we get the transactionID like we do on the EditRequestPage [here](https://github.com/Expensify/App/blob/eb670822abb92e1cd6a7834f2ccda629854a339e/src/pages/EditRequestPage.js#L70-L71)?

Reply:
Also, I'm seeing an interesting technique on that page where they call withOnyx twice, which I think might be a way to use Onyx props in withOnyx. https://github.com/Expensify/App/blob/eb670822abb92e1cd6a7834f2ccda629854a339e/src/pages/EditRequestPage.js#L193-L205

Reply:
@neil-marcellini The method used on the EditRequestPage is an anti-pattern that I'd like to remove. I've created a cleanup issue here with more details: https://github.com/Expensify/App/issues/27262

I also think the double `withOnyx()` is a mistake.